core/dbt/task/retry.py

```diff
@@ -28,6 +32,7 @@
     "defer_state",
     "deprecated_state",
     "target_path",
+    "warn_error",
```

Comment:
@graciegoheen how do we feel about `warn_error` being a thing that's being carried over?
Current behavior:
- `dbt run --warn-error` failed because of some warning seen as error
- 'dbt retry' will not treat those warnings as error
- 'dbt retry --warn-error' will be the way to still treat warnings as error.
This is the previous behavior we have. Do we think this is good? or do we want to change it?

Reply:
@peterallenwebb I think we talked about this became a behavior change in my version. But in the latest version this is actually not.

Reply:
> - dbt run --warn-error failed because of some warning seen as error
> - 'dbt retry' will not treat those warnings as error

This is surprising to me. It ties back to the desire to have a [consistent](https://github.com/dbt-labs/dbt-core/issues/9325) policy here. I would assume that **all** flags passed to the original command are pulled through to `dbt retry`. Is this a specific exception for `--warn-error`? Ultimately, if we're worried about breaking something in a backport, we could address this for `1.8`.

Reply:
So `warn-error` is a flag, so if we don't ignore it in retry there's actually no way to specify `no-warn-error` in retry.


> dbt run --warn-error failed because of some warning seen as error
'dbt retry' will not treat those warnings as error

Just to be super clear this is the current behavior of retry.